services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py

```diff
@@ -98,156 +97,154 @@ async def is_healthy(self, dynamic_sidecar_endpoint: str) -> bool:
         except httpx.HTTPError:
             return False
 
+    @log_decorator(logger=logger)
     async def containers_inspect(self, dynamic_sidecar_endpoint: str) -> Dict[str, Any]:
         """
         returns dict containing docker inspect result form
         all dynamic-sidecar started containers
         """
         url = get_url(dynamic_sidecar_endpoint, f"/{self.API_VERSION}/containers")
-        try:
-            response = await self._client.get(url=url)
-            if response.status_code != status.HTTP_200_OK:
-                message = (
-                    f"error during request status={response.status_code}, "
-                    f"body={response.text}"
-                )
-                logging.warning(message)
-                raise DynamicSidecarNetworkError(message)
-
-            return response.json()
-        except httpx.HTTPError:
-            log_httpx_http_error(url, "GET", traceback.format_exc())
-            raise
 
+        response = await self._client.get(url=url)
+        if response.status_code != status.HTTP_200_OK:
+            raise DynamicSidecarUnexpectedResponseStatus(response)
+
+        return response.json()
+
+    @log_decorator(logger=logger)
     async def containers_docker_status(
         self, dynamic_sidecar_endpoint: str
     ) -> Dict[str, Dict[str, str]]:
         url = get_url(dynamic_sidecar_endpoint, f"/{self.API_VERSION}/containers")
-        try:
-            response = await self._client.get(url=url, params=dict(only_status=True))
-            if response.status_code != status.HTTP_200_OK:
-                logging.warning(
-                    "error during request status=%s, body=%s",
-                    response.status_code,
-                    response.text,
-                )
-                return {}
-
-            return response.json()
-        except httpx.HTTPError:
-            log_httpx_http_error(url, "GET", traceback.format_exc())
-            raise
 
+        response = await self._client.get(url=url, params=dict(only_status=True))
+        if response.status_code != status.HTTP_200_OK:
+            logging.warning(
```

Comment:
one question, why you do not raise the ```DynamicSidecarUnexpectedResponseStatus``` here?

Reply:
Because I am interested in the single state of each docker container. Until they start it is fine to simply return an empty dictionary. When the container is finally available in some state it will return it.
At the time I think, I preferred to return a simpler object to deal with.